src/main/java/mate/academy/spring/controller/CustomGlobalExceptionHandler.java

```diff
@@ -0,0 +1,55 @@
+package mate.academy.spring.controller;
+
+import java.time.LocalDateTime;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import mate.academy.spring.exception.DataProcessingException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(MethodArgumentNotValidException e,
+                                                                  HttpHeaders headers,
+                                                                  HttpStatus status,
+                                                                  WebRequest request) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now().toString());
+        body.put("status", status.value());
+        List<String> errorMessages = e.getBindingResult()
+                .getAllErrors()
+                .stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errorMessages);
+        return new ResponseEntity<>(body, headers, status.value());
+    }
+
+    @ExceptionHandler(DataProcessingException.class)
+    protected ResponseEntity<Object> handleDataProcessingException(DataProcessingException e,
+                                                                               HttpHeaders header) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now().toString());
+        body.put("status", HttpStatus.INTERNAL_SERVER_ERROR.value());
+        body.put("errors", e.getMessage());
+        return new ResponseEntity<>(body, header, HttpStatus.INTERNAL_SERVER_ERROR.value());
+    }
```

Comment:
almost :)

```suggestion
    protected ResponseEntity<Object> handleDataProcessingException(DataProcessingException e) {
        Map<String, Object> body = new LinkedHashMap<>();
        body.put("timestamp", LocalDateTime.now().toString());
        body.put("status", HttpStatus.INTERNAL_SERVER_ERROR.value());
        body.put("exception", e.getMessage());
        return new ResponseEntity<>(body, HttpStatus.INTERNAL_SERVER_ERROR);
    }
```

this exception is thrown by repository method
our dao layer dosen't send any http responses to service, so there can't be any httpHeader here :)

Reply:
:( l tried